internal/cmd/controller/agentmanagement/controllers/cluster/import.go

```diff
@@ -77,47 +87,60 @@ func RegisterImport(
 	clusters.OnChange(ctx, "import-cluster", h.OnChange)
 	fleetcontrollers.RegisterClusterStatusHandler(ctx, clusters, "Imported", "import-cluster", h.importCluster)
 	config.OnChange(ctx, h.onConfig)
+
+	clustersCache := clusters.Cache()
+	clustersCache.AddIndexer(clusterForKubeconfigSecretIndexer, func(cluster *fleet.Cluster) ([]string, error) {
+		if cluster == nil || len(cluster.Spec.KubeConfigSecret) == 0 {
+			return []string{}, nil
+		}
+		secretKey := getKubeConfigSecretNS(cluster) + "/" + cluster.Spec.KubeConfigSecret
+		return []string{secretKey}, nil
+	})
+	secrets.OnChange(ctx, "kubeconfig-secrets-watch", func(key string, secret *corev1.Secret) (*corev1.Secret, error) {
+		clusters, err := clustersCache.GetByIndex(clusterForKubeconfigSecretIndexer, key)
```

Comment:
Does this mean that more than one clusters can share the same key, therefore the same `KubeConfigSecret`? :thinking:

Reply:
It's technically possible, yes, because we don't see any restriction on the unicity of `Spec.KubeConfigSecret` across Clusters, although it may not make sense at all in the practice. Most of the times, `clusters` will only have a length of 0 or 1.

Reply:
Indeed, my understanding is that a length of more than 1 would, given the way the index key is built, imply multiple clusters sharing a namespace, which would cause other issues for Fleet.